rust/worker/src/execution/operators/metadata_filtering.rs

```diff
@@ -120,618 +418,135 @@ impl Operator<MetadataFilteringInput, MetadataFilteringOutput> for MetadataFilte
-                WhereDocumentOperator::Contains => {
-                    // Note matching_contains is sorted (which is needed for correctness)
-                    // because materialized log record is sorted by offset id.
-                    let mut matching_contains = vec![];
-                    // Upstream sorts materialized records by offset id so matching_contains
-                    // will be sorted.
-                    // Note: Uncomment this out when adding FTS support for queries
-                    // containing _ or %. Currently, we disable such scenarios in tests
-                    // for distributed version.
-                    // Emulate sqlite behavior. _ and % match to any character in sqlite.
-                    // let normalized_query = query.replace("_", ".").replace("%", ".");
-                    // let re = Regex::new(normalized_query.as_str()).unwrap();
-                    for (record, _) in mat_records.iter() {
-                        if record.final_operation == MaterializedLogOperation::DeleteExisting {
-                            continue;
-                        }
-                        match record.merged_document_ref() {
-                            Some(doc) => {
-                                /* if re.is_match(doc) { */
-                                if doc.contains(query) {
-                                    matching_contains.push(record.offset_id as i32);
-                                }
-                            }
-                            None => {}
-                        };
-                    }
-                    return matching_contains;
-                }
-                WhereDocumentOperator::NotContains => {
-                    todo!()
-                }
-            }
+
+        let materialized_log_oids = match filterd_log_oids {
+            Include(rbm) => rbm,
+            Exclude(rbm) => metadata_log_reader.active_domain() - rbm,
         };
-        // fts_result will be sorted by offset id.
-        let fts_result = match &input.where_document_clause {
-            Some(where_doc_clause) => {
-                match process_where_document_clause_with_callback(where_doc_clause, &cb) {
-                    Ok(res) => {
-                        let ids_as_u32: Vec<u32> =
-                            res.into_iter().map(|index| index as u32).collect();
-                        tracing::info!(
-                            "Filtered {} results from log based on where document filtering",
-                            ids_as_u32.len()
-                        );
-                        Some(ids_as_u32)
-                    }
-                    Err(e) => {
-                        tracing::error!("Error filtering logs based on where document {:?}", e);
-                        return Err(MetadataFilteringError::MetadataFilteringIndexError(e));
-                    }
+
+        let filtered_compact_oids = if let Some(clause) = input.where_clause.as_ref() {
```

Comment:
nit: add comment here that this takes care of filtering the deletes on the log from the metadata segment